src/exec/mod.rs

```diff
@@ -224,6 +242,61 @@ impl ArgumentTemplate {
             ArgumentTemplate::Text(ref text) => OsString::from(text),
         }
     }
+
+    /// Replace the path separator in the input with the custom separator string. If path_separator
+    /// is None, simply return a borrowed Cow<OsStr> of the input. Otherwise, the input is
+    /// interpreted as a Path and its components are iterated through and re-joined into a new
+    /// OsString.
+    fn replace_separator<'a>(path: &'a OsStr, path_separator: Option<&str>) -> Cow<'a, OsStr> {
+        // fast-path - no replacement necessary
+        if path_separator.is_none() {
+            return Cow::Borrowed(path);
+        }
+
+        let path_separator = path_separator.unwrap();
+        let mut out = OsString::with_capacity(path.len());
```

Comment:
will only be enough if the path separator is on character, right? But okay, that's a strange corner case where we can live additional allocations, I guess.

Reply:
Yeah, I think the Vec behind OsString grows by doubling in size so there's probably only one or two extra allocations for multi-character separators. The only way to know exactly how big out needs to be is by iterating & counting the path components beforehand, so just letting OsString automatically resize seems like a reasonable thing to do.